webSites/webSites-war/src/main/java/org/silverpeas/components/websites/servlets/design/SiteDesignActionHandler.java

```diff
@@ -0,0 +1,648 @@
+      return (avant + parseCodeSupprImage(scc, apres, request, settings, currentPath));
+    }
+  }
+
+  private String parseCodeSupprHref(WebSiteSessionController scc, String code, String currentPath) {
+    String avant;
+    String apres;
+    int index;
+    String href = "<A href=\""; /* longueur de chaine = 9 */
+
+    String finChemin;
+    String fichier;
+    String deb;
+    String theReturn = "";
+
+    index = code.indexOf(href);
+    if (index == -1) {
+      theReturn = code;
+    } else {
+
+      avant = code.substring(0, index + 9);
+
+      apres = code.substring(index + 9);
+
+      if (apres.startsWith("http://")) {
+        /* external link */
+        theReturn = avant + parseCodeSupprHref(scc, apres, currentPath);
+      } else if (apres.startsWith("ftp://")) {
+        /* external link */
+        theReturn = avant + parseCodeSupprHref(scc, apres, currentPath);
+      } else if (apres.startsWith("rr:")) {
+        /* already in a relative form */
+        apres = apres.substring(3);
+        theReturn = avant + parseCodeSupprHref(scc, apres, currentPath);
+      } else if (apres.startsWith("aa:")) {
+        /* the link is in an absolute form: convert it into the relative form */
+        /* finChemin = rep/coucou.html">... */
+        finChemin = code.substring(index + 12);
+        int indexGuillemet = finChemin.indexOf("\"");
+        /* absolute = rep/coucou.html */
+        String absolute = finChemin.substring(0, indexGuillemet);
+        /* apres = ">... */
+        apres = finChemin.substring(indexGuillemet);
+        int indexSlash = absolute.lastIndexOf("\\");
+        if (indexSlash == -1) {
+          // pas d'arborescence, le fichier du lien est sur la racine
+          fichier = absolute;
+          deb = "";
+        } else {
+          /* fichier = coucou.html */
+          fichier = absolute.substring(indexSlash + 1);
+          deb = absolute.substring(0, indexSlash);
+        }
+        List<String> tab = buildTab(deb + "/");
+        // dans ce tableau il manque l'id
+
+        /* cheminContexte = id/rep */
+        int longueur = scc.getComponentId().length();
+        int index2 = currentPath.lastIndexOf(scc.getComponentId());
+        String chemin = currentPath.substring(index2 + longueur);
+
+        chemin = chemin.substring(1);
+        chemin = removeDoubleAntiSlash(chemin);
+        String cheminContexte = chemin;
+        List<String> tabContexte = buildTab(cheminContexte + "/");
+        /* ajoute l'id dans le premier tableau */
+        tab.add(0, tabContexte.get(0));
+
+        /* tabCommun = [id | rep] */
+        List<String> tabCommun = sortCommun(tabContexte, tab);
+
+        /* reste = vide */
+        String reste = sortReste(tab, tabCommun);
+
+        /* nbPas = 0 */
+        int nbPas = tabContexte.size() - tabCommun.size();
+        StringBuilder relatif = new StringBuilder();
+        for (int i = 0; i < nbPas; i++) {
+          relatif.append("../");
+          i++;
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Refactor the code in order to not assign to this loop counter from within the loop body. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://integration.silverpeas.org/sonar/coding_rules#rule_key=squid%3AForLoopCounterChangedCheck)